framework/components/s3provider/interfaces.go

```diff
@@ -0,0 +1,19 @@
+package s3provider
+
+// Provider is the interface that wraps S3 interaction methods.
+type Provider interface {
+	GetEndpoint() string
+	GetDockerEndpoint() string
+	GetConsoleURL() string
+	GetSecretKey() string
+	GetAccessKey() string
+	GetBucket() string
+	GetRegion() string
+	GetOutput() *Output
```

Comment:
nit: isn't it more idiomatic in go to have these functions called `Output` instead of `GetOutput`?